source/main/physics/BeamFactory.cpp

```diff
@@ -1671,28 +1671,28 @@ std::shared_ptr<RigDef::File> ActorManager::FetchActorDef(const char* filename,
     }
 }
 
-int ActorManager::CountActorsInternal() const
+std::vector<Actor*> ActorManager::GetActors() const
 {
-    int count = 0;
+    std::vector<Actor*> actors;
     for (int t = 0; t < m_free_actor_slot; t++)
     {
         if (m_actors[t] != nullptr)

```

Comment:
This mechanism is an abomination, and I think we're ready to get rid of it soon.

Reply:
I always wanted to get rid of that. If I remember correctly the main issue were the unique material names, which were based off the truck array indices.